src/geode/geosciences/well_dev_input.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/geosciences/private/well_dev_input.h>
+
+#include <fstream>
+
+#include <absl/strings/str_split.h>
+
+#include <geode/basic/attribute_manager.h>
+
+#include <geode/geometry/point.h>
+
+#include <geode/mesh/builder/edged_curve_builder.h>
+#include <geode/mesh/core/edged_curve.h>
+
+#include <geode/geosciences/private/utils.h>
+
+namespace
+{
+    struct HeaderData
+    {
+        absl::string_view name{ "unknown" };
```

Comment:
This might be undefined behavior since "unknown" is not owned by anyone. This is dangerous